src/mjolnir/osmway.cc

```diff
@@ -13,6 +14,50 @@ constexpr uint32_t kMaxNodesPerWay = 65535;
 constexpr uint8_t kUnlimitedOSMSpeed = std::numeric_limits<uint8_t>::max();
 constexpr float kMaxOSMSpeed = 140.0f;
 
+const std::regex kFloatRegex("\\d+\\.(\\d+)");
+
+/**
+ * For levels we use 7-bit varint encoding, with arbitrary precision stored separately.
+ *
+ * @param lvl       the level to encode
+ * @param precision the precision with which to encode the level value
+ *
+ * @return a string containing the encoded value
+ */
+std::string encode_level(const float lvl, const int precision = 0) {
+  std::string encoded;
+  // most of the time precision will be zero and level values will be lower
+  // than 4191 (even in the case of higher precision)
+  encoded.reserve(2);
+
+  int val = static_cast<int>(lvl * pow(10, precision));
+  const bool negative = val < 0;
```

Comment:
The previous CI fails arose from a `static_cast` from int to unsigned int back to int. According to C++17 (§7.8/3 (Integral Conversions):

> If the destination type is signed, the value is unchanged if it can be represented in the destination type;
otherwise, the value is implementation-defined 

So instead, we perform a more manual conversion (taken from [another polyline implementation](https://github.com/vahancho/polylineencoder/blob/07db1476f369ff64c9bc2778021b064710847764/src/polylineencoder.h#L194-L211)

Reply:
Edit: after review with @kevinkreiser we decided to just use `reinterpret_cast` instead